.github/workflows/build.yml

```diff
@@ -43,4 +43,4 @@ jobs:
 
       - name: Test
         run: |
-          nosetests --with-doctest --doctest-extension=rst
+          PYTHONPATH=. py.test --doctest-modules --doctest-glob='*.rst'
```

Comment:
This should be able to run without PYTHONPATH=

`pytest --doctest-modules --doctest-glob='*.rst'`

This works locally for me on Windows

Reply:
I am quite certain I tried it (only with py.test, not pytest) and that it failed on Ubuntu

Reply:
I see.  Yes, you're right.  It is still failing.

The only difference I see at this point between this branch and the documentation is that the documentation installs pytest in the same stage that it executes under.  That could be the last possible difference, but I'm not sure why it would matter at all (I'd guess nose and pytest install differently).

I am happy to merge this change and then work on getting the build action working again separately.  The changes themselves to the tests are good :)